### PR TITLE
fix(core): support docker engine v20.10.0-beta1

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Running an image by only providing the hash fails on docker engine v20.10.0-beta1 with invalid repository name.

```
docker run --rm b92402b29db56f1bbace74c369bedef5ee296a76fd8545426255247da70ce21a
docker: Error response from daemon: invalid repository name (b92402b29db56f1bbace74c369bedef5ee296a76fd8545426255247da70ce21a), cannot specify 64-byte hexadecimal strings.
```

Using `docker run --rm sha256:b92402b29db56f1bbace74c369bedef5ee296a76fd8545426255247da70ce21a` instead works as expected.

I haven't been able to pinpoint the exact change yet as this seems not to be mentioned in https://github.com/docker/docker-ce/blob/0fc7084265b3786a5867ec311d3f916af7bf7a23/CHANGELOG.md

Created an issue with docker to clarify whether this is a regression or a planned change https://github.com/docker/cli/issues/2815

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/lib/bundling.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class BundlingDockerImage {
     // different every time the Docker layer cache is cleared, due primarily to
     // timestamps.
     const hash = FileSystem.fingerprint(path, { extraHash: JSON.stringify(options) });
-    return new BundlingDockerImage(match[1], hash);
+    return new BundlingDockerImage(`sha256:${match[1]}`, hash);
```
Comment (jogold) [inline packages/@aws-cdk/core/lib/bundling.ts:132]: Can we change the regex on L120 instead? and use `match[0]` then.

https://github.com/aws/aws-cdk/blob/867221b22df0196d6be7ff4d0eca96408e4775b0/packages/%40aws-cdk/core/lib/bundling.ts#L120

Comment (johanneswuerbach) [reply packages/@aws-cdk/core/lib/bundling.ts]: Sure, updated.

**Comment by johanneswuerbach** [comment]
Anything I can do to move this forward? Otherwise the CDK is broken on the Docker for Mac Edge channel when you build assets in docker.

File: packages/@aws-cdk/core/lib/bundling.ts [hunk 2]
```diff
@@ -117,7 +117,7 @@ export class BundlingDockerImage {
 
     const docker = dockerExec(dockerArgs);
 
-    const match = docker.stdout.toString().match(/sha256:([a-z0-9]+)/);
+    const match = docker.stdout.toString().match(/(sha256:[a-z0-9]+)/);
```
Comment (jogold) [inline packages/@aws-cdk/core/lib/bundling.ts:120]:
```suggestion
    const match = docker.stdout.toString().match(/sha256:[a-z0-9]+/);
```

**Comment by jogold** [comment]
@rix0rrr or @eladb can you take a look here? LGTM

**Review by eladb — APPROVED** [review]

**Comment by sirvio** [comment]
hmm, seems not to work for me yet. Still receive "invalid repository name ..... cannot specify 64-byte hexadecimal strings" when using docker engine 20.10.0-beta1 for mac and cdk 1.71.0.

**Comment by jogold** [comment]
> hmm, seems not to work for me yet. Still receive "invalid repository name ..... cannot specify 64-byte hexadecimal strings" when using docker engine 20.10.0-beta1 for mac and cdk 1.71.0.

This fix will be included in the next release.

**Pull request merged**